src/main/java/EntropySolver.java

```diff
@@ -0,0 +1,129 @@
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static com.google.common.collect.Maps.newHashMap;
+import static com.google.common.collect.Maps.newLinkedHashMap;
+import static com.google.common.collect.Sets.newHashSet;
+import static java.lang.Math.log;
+
+// https://gregcameron.com/infinite-wordle/
+
+@Slf4j
+public class EntropySolver {
+    private static final BufferedReader READER = new BufferedReader(new InputStreamReader(System.in));
+    private static final Set<String> ACCEPTED = newHashSet();
+
+    public static void main(String[] args) {
+        Set<String> solutions = newHashSet();
+        processInput(solutions);
+
+        // precomputed best at first step
+        log.info("Best Word: {}", "soare");
+        filterRemainingWords(solutions, "soare", getOutcome());
+        log.info("{} - {}", solutions.size(), solutions.size() < 50 ? solutions : "");
+
+        while (solutions.size() > 1) {
+            Map<String, Double> entropyMap = newHashMap();
+            ACCEPTED.forEach(potentialInput -> {
+                Map<Integer, Integer> patterns = newHashMap();
+                solutions.forEach(solution -> patterns.merge(calculatePattern(potentialInput, solution), 1, Integer::sum));
+                entropyMap.put(potentialInput, getEntropy(patterns));
+            });
+
+            String bestWord = getBestWord(entropyMap);
+            log.info("Best Word: {}", bestWord);
+            filterRemainingWords(solutions, bestWord, getOutcome());
+            log.info("{} - {}", solutions.size(), solutions.size() < 100 ? solutions : "");
```

Comment:
cyann- refactor all instance of ? : if else logic to more readable alternative. Use if else not ? :